webapp/app.py

```diff
@@ -0,0 +1,152 @@
+from inspect import getsourcefile
+# import from parent dir according to https://stackoverflow.com/a/9331002
+import os.path as path, sys
+current_dir = path.dirname(path.abspath(getsourcefile(lambda:0)))
+sys.path.insert(0, current_dir[:current_dir.rfind(path.sep)])
+import midi_to_part_mp3s
+sys.path.pop(0)
+
+
+app = Flask(__name__)
+
+FILE: str = 'file'
+SOPRANO: str = 'soprano'
+ALTO: str = 'alto'
+TENOR: str = 'tenor'
+BASS: str = 'bass'
+INSTRUMENT: str = 'instrument'
+MUSICXML: str = 'musicxml'
+OUTPUT: str = 'output'
+
+TEMP_FOLDER: str = 'tmp'
+OUTPUT_FOLDER: str = 'out'
+FILENAME: str = 'part-mp3s.zip'
+
+
+@app.route("/")
+def index():
+    return render_template('index.html')
+
+
+@app.route("/midi-info", methods=['POST'])
+def midi_info():
+    random_folder_name = _check_tmp_folder()
+    midi_file_name = _get_musicxml_file(random_folder_name)
+    print(midi_file_name)
+    score: stream.Score = converter.parse(midi_file_name)
+    print(score)
+    musicxml_filename = "%s/%s" % (random_folder_name, 'score.musicxml')
+    score.write(fp=musicxml_filename)
+    xml_content = None
+    with open(musicxml_filename) as file_stream:
+        xml_content = file_stream.read()
+    response_dict = {}
+    response_dict['musicxml'] = xml_content
+    response_dict['parts'] = []
+    part_stream = score.parts.stream()
+    for part in part_stream:
+        response_dict['parts'].append(part.id)
+    return json.dumps(response_dict)
+
+
+@app.route("/download", methods=['POST'])
+def midi_to_mp3():
+    random_folder_name = _check_tmp_folder()
+    soprano_tracks = request.form.getlist(SOPRANO)
+    alto_tracks = request.form.getlist(ALTO)
+    tenor_tracks = request.form.getlist(TENOR)
+    bass_tracks = request.form.getlist(BASS)
+    instrument_id = request.form.get(INSTRUMENT)
+    music_xml = request.form.get(MUSICXML)
+    midi_file_path = _save_musicxml_file(random_folder_name, music_xml)
+
+    print("soprano-tracks: %s, alto_tracks: %s, tenor_tracks: %s, bass_tracks: %s, instrument_id: %s, music_xml: %s characters" % (
+        soprano_tracks, alto_tracks, tenor_tracks, bass_tracks, instrument_id, len(
+            music_xml)
+    ))
+
+    argument_list: List[str] = [midi_file_path]
+    _list_append_track_id(argument_list, SOPRANO, soprano_tracks)
+    _list_append_track_id(argument_list, ALTO, alto_tracks)
+    _list_append_track_id(argument_list, TENOR, tenor_tracks)
+    _list_append_track_id(argument_list, BASS, bass_tracks)
+    _single_append_argument(argument_list, INSTRUMENT, instrument_id)
+
+    output_folder = '%s/%s' % (random_folder_name, OUTPUT_FOLDER)
+    _single_append_argument(argument_list, OUTPUT, output_folder)
+    print('Starting midi_to_part_mp3s with parameter list: %s' % argument_list)
+    midi_to_part_mp3s.main(argument_list)
+
+    zip_file_name = '%s/%s' % (random_folder_name, FILENAME)
+    print ('Creating zip file: %s' % zip_file_name)
+    with ZipFile(zip_file_name, 'w') as myzip:
```

Comment:
Maybe all this could be encapsulated in `send_zip_file`?

Reply:
Ok, I will change that.